syntax = "proto3";

package params;

service WeightsPasser {
    //rpc sendWeights (WeightsToServer) returns (WeightsToClient) {}
    //rpc sendWeights (stream Parameters) returns (stream Parameters) {}
    rpc sendWeights (Parameters) returns (stream Parameters) {}
}

message Parameters {
    bytes parameters = 1;
    string tensor_type = 2;
}

/*message WeightsToServer {
    //repeated bytes parameters = 1;
    //swap back to repeated bytes when protobuf's working
    bytes parameters = 1;
    //string parameters = 1;
    string tensor_type = 2;
}

message WeightsToClient {
    string reply = 1;
}*/
